### PR TITLE
feat: npm release build fix + readme files updated

### PR DESCRIPTION
**Pull request by tarun-khanna**
# Description

1. Fixes the build fail issue on release pipeline. 
Screenshot 2024-01-24 at 1 38 42 PM

2. Added root readme file that links to readme files of `i18nify-js` and `i18nify-react`

3. Added basic readme file for `i18nify-react`

List the main changes made in this pull request.

| Title                  | --- |
| ---------------------- | --- |
| JIRA link              | https://razorpay.atlassian.net/browse/ATLAS-107  |
| Slack thread (if any)  | NA  |
| Product spec           | NA  |
| Tech spec/One-pager    | NA  |
| Bundle Size Difference | NA  |
| Fixes Issue            | NA  |

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] This change is a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Is your change backward compatible ?
- [ ] Tested on major browsers (Chrome, Firefox, Safari, IE) ?
- [ ] Tested in a consumer application(s) ?

## Additional Notes

Any additional information that would be helpful for the reviewer.

## Checklist:

- [x] Add Jira ID(s) in PR title and in the description?
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code is written in TypeScript and follows the coding standards of the project.
- [x] I have added relevant documentation and updated the README if necessary.
- [x] My commit messages are clear and follow the project's commit message conventions.
- [ ] Is any external library added?
- [x] My changes do not introduce any new runtime errors or warnings.
- [ ] All relevant unit tests have been added/updated in my PR for the change.
- [ ] Any Screenshots (mobile & desktop) required for PR? If yes, have you added the respective screenshots ?
- [ ] Any manual dev testing done by you on beta/func environment?
- [ ] Reviewer added (SLA 2 days)

## Reviewer Checklist

- [ ] Sufficient QA/Dev-Testing is done with proof (test cases list)
- [ ] Ensure that the change in bundle size falls within the acceptable range.

---

## PR Title Format

Format: `<type>: <subject>`

Types can be as follows:

- `feat`: (new feature for the user, not a new feature for build script)
- `fix`: (bug fix for the user, not a fix to a build script)
- `docs`: (changes to the documentation)
- `style`: (formatting, missing semi colons, etc; no production code change)
- `refactor`: (refactoring production code, eg. renaming a variable)
- `test`: (adding missing tests, refactoring tests; no production code change)
- `chore`: (updating grunt tasks etc; no production code change)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -23,6 +23,8 @@ jobs:
           node-version: 20.3.1
       - name: Install dependencies
         run: yarn
+      - name: run build
```
Comment (avinashdvv) [inline .github/workflows/release.yml:26]: suggestion: please update this naming as react build

File: packages/i18nify-js/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ And hey, hang tight—I’ll break down each of these enchanting modules in the
 
 ## Install
 
-`yarn add i18nify`
+`yarn add @razorpay/i18nify-js`
```
Comment (avinashdvv) [inline packages/i18nify-js/README.md:11]: question: can't we install this directly `yarn add 18nify-js` like this?

**Review by RgnDunes — APPROVED** [review]
Looks good to me

File: packages/i18nify-react/README.md [hunk 3]
```diff
@@ -1 +1,90 @@
 # i18nify-react
+
+Welcome to i18nify-react, your new best friend for seamlessly managing i18n state in React applications! 🎉 The i18nify-react Context Provider is a simple and flexible solution for managing internationalization (i18n) state in React applications. This provider utilizes the @razorpay/i18nify-js library to handle i18n state and exposes a React Context for efficient state management across the application.
+
+## Features
+
+- **I18nProvider:** A custom React Context provider that effortlessly spreads i18n state updates across your component tree.
+- **useI18nContext Hook:** Fetch values exposed by the `I18nProvider` using this handy custom React hook.
+- **Easy Integration:** Integrate i18n state management into your React app with style and ease.
```
Comment (avinashdvv) [inline packages/i18nify-react/README.md:9]: change:  `I18nProvider` explaining tech details, `Easy Integration` explaining the operational details, clubbing them together seems confusing.

With `I18nProvider` and `useI18nContext` you can easily integrate i18n state management into your app. 

I18nProvider: A custom React Context provider that effortlessly spreads i18n state updates across your component tree.
useI18nContext: Fetch values exposed by the `I18nProvider` using this handy custom React hook.

something like this.

explain like this
1. What does it do
2. How does it do
3. What changes developer required todo make it work effectively

**Review by avinashdvv — CHANGES_REQUESTED** [review]

File: packages/i18nify-js/README.md (hunk 2 above)
Comment (tarun-khanna) [reply packages/i18nify-js/README.md:11]: No, the package is named as `@razorpay/i18nify-js`. This must have been missed in the documentation before.

File: packages/i18nify-react/README.md (hunk 3 above)
Comment (tarun-khanna) [reply packages/i18nify-react/README.md:9]: This is just a starter readme that I've added from this [PR](https://github.com/razorpay/i18nify/pull/59) for an empty readme file. 
@RgnDunes will take some time to refine the items as suggested before. 

FYI, Above questions are addressed in below sections of the readme.
1. Introduction and Features
2. Installation
3. Usage

Comment (avinashdvv) [reply packages/i18nify-react/README.md:9]: Got it, can you please create the JIRA ticket for readme refinement and add that here, i will approv the PR.

Comment (avinashdvv) [reply packages/i18nify-react/README.md:9]: Based on this, https://razorpay.slack.com/archives/C05RBSKC6KD/p1706089761213119?thread_ts=1706083683.738819&cid=C05RBSKC6KD

I am considering this comment as addressed

**Review by avinashdvv — APPROVED** [review]
LGTM

**Pull request merged**